lib/pbench/server/api/resources/users_api.py

```diff
@@ -58,13 +57,9 @@ def get(self, target_username: str) -> Response:
                 "last_name": token_payload.get("family_name"),
                 "email": token_payload.get("email"),
             }
-        elif current_user.is_admin():
-            target_user = User.query(username=target_username)
-            if target_user:
-                response_object = target_user.get_json()
-            else:
-                abort(HTTPStatus.NOT_FOUND, message="Target user not found")
         else:
+            # If the decoded token does not contain the targeted username,
+            # return FORBIDDEN
```

Comment:
Yeah, I did realize yesterday that if keep this, we can not ask for the `target_username`. This acts strictly like a "token decryption" functionality on the server where the requester gives the pbench server a token and pbench server decrypts it and returns what it wants to return from the token payload.

Reply:
No, this has nothing to do with `/api/v1/endpoints` ... that's global *server* API configuration, and it's not authenticated anyway.

If we're providing an API to decrypt the authentication token, it's a dedicated API, maybe `GET /api/v1/me` 😉 

I certainly don't object to a `GET /datasets/list?mine`, either, and that might be convenient... but it's also rather a bit weird.

Reply:
> If we're providing an API to decrypt the authentication token, it's a dedicated API, maybe `GET /api/v1/me`

So do we want to retain this or remove it? and deal with the fallout later

> I certainly don't object to a GET /datasets/list?mine, either, and that might be convenient... but it's also rather a bit weird.

We could change the default behavior of `_build_sql_query` functionality to return only the private dataset for the authorized user even if no `owner id` is specified. If the requester wants to see the public datasets as well then they would have to explicitly ask for that in the request like `include_public` or something.

Reply:
No! The current unqualified "show me everything I'm allowed to see" is perfect.

The problem we're now introducing is that the primary client doesn't know the identity of the authenticated user, and in one context wants to filter on that value. So if we're not going to rely on the client figuring out the user's name on its own, an anonymous filter like `mine` (or `client`, or ...) makes sense.

Reply:
Oh, but there is another issue if the `owner` parameter is not provided the `_check_authorization` would fail for APIs with ApiAuthorizationType as `USER_ACCESS` that would also need to be fixed.

Reply:
> No! The current unqualified "show me everything I'm allowed to see" is perfect.
> 
> The problem we're now introducing is that the primary client doesn't know the identity of the authenticated user, and in one context wants to filter on that value. So if we're not going to rely on the client figuring out the user's name on its own, an anonymous filter like `mine` (or `client`, or ...) makes sense.

Okay, so do we want to remove this "token decryption" functionality from the server that the client can use to know about the user's identity?

Reply:
Let's just remove it. I'll implement a new filter. Maybe I'll just call it "mine" unless someone has a better idea. The OIDC changes are still in a branch, so you won't be breaking the dashboard Overview page until the branch merges ... so timing isn't critical.

Reply:
Alrighty, I'll address it in the next commit

Reply:
> I'll just call it "mine" unless someone has a better idea.

`mine` is good by me.

> you won't be breaking the dashboard Overview page until the branch merges

I'm still confused on how the changes in this PR break will the Dashboard (other than the facts that the user won't be able to "log into the Pbench Server" or be able to provide a valid token.... 😮).  That is, I don't see how the removal of the `GET /user` resource will actually cause any problems, because the Dashboard won't get far enough to notice....  🤣

Reply:
Sure, we don't need this until we cut the dashboard over to using Keycloak, but that's the whole point of this exercise. (One step at a time.) Once we get there, we need a way for the Overview page to isolate the authenticated user's datasets for curation.